File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/inherit_missing_runme.lua
require("import")	-- the import fn
import("inherit_missing")	-- import lib
im=inherit_missing

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

bar = im.Bar()
spam = im.Spam()

assert(im.do_blah(bar) == "Bar::blah")
assert(im.do_blah(spam) == "Spam::blah")

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/inherit_missing_runme.go
package main

import "./inherit_missing"

func main() {
	a := inherit_missing.New_Foo()
	b := inherit_missing.NewBar()
	c := inherit_missing.NewSpam()

	x := inherit_missing.Do_blah(a)
	if x != "Foo::blah" {
		panic(x)
	}

	x = inherit_missing.Do_blah(b)
	if x != "Bar::blah" {
		panic(x)
	}

	x = inherit_missing.Do_blah(c)
	if x != "Spam::blah" {
		panic(x)
	}

	inherit_missing.Delete_Foo(a)
}
